R/appDependencies.R

```diff
@@ -8,6 +8,27 @@
 #' and adds all recursive dependencies to create a complete manifest of
 #' package packages need to be installed to run the app.
 #'
+#' # Dependency discovery
+#'
+#' rsconnect use one of three mechanisms to find which packages your application
+#' uses:
+#'
+#' 1. If `renv.lock` is present, it will use the versions and sources defined in
+#'    that file. If you're using the lockfile for some other purpose and
+#'    don't want it to affect deployment, add `renv.lock` to `.rscignore`.
+#'
+#' 2. Otherwise, rsconnect will call `renv::snapshot()` to find all packages
+#'    used by your code. If you'd instead prefer to only use the packages
+#'    declared in a `DESCRIPTION` file, run
+#'    `renv::settings$snapshot.type("explicit")` to activate renv's "explicit"
```

Comment:
`renv::snapshot()` does accept a `type` argument so this would be straightforward to wire up, at least.

Reply:
My sense is that once we enable support for renv lockfile, we'll get quite a few feature requests so we might as well just let them accumulate for a while to get a sense for what people actually want, rather than doing too much prospectively.